File: 937_Reorder_Data_in_Log_Files.cpp
class Solution {
public:
    vector<string> reorderLogFiles(vector<string>& logs) {
        stable_sort(logs.begin(),logs.end(), [](const string& a, const string& b) -> bool {
            int i = a.find(' '), j = b.find(' ');
            
            if (!isdigit(a[i+1]) && !isdigit(b[j+1])) {
                string s1 = a.substr(i+1), s2 = b.substr(j+1);
                if(s1 == s2) return a.compare(b) < 0;
                return s1.compare(s2) < 0;
            }
            
            return !isdigit(a[i+1]);
        });
        
        return logs;
    }
};

File: 937_Reorder_Data_in_Log_Files.py
class Solution:
    def reorderLogFiles(self, logs: List[str]) -> List[str]:
        output = []
        digits = []
        letters = {}
        
        for log in logs:
            if log.split()[1].isnumeric():
                digits.append(log)
            else:
                key = ' '.join(log.split()[1:])
                if key not in letters:
                    letters[key] = []
                letters[key].append(log.split()[0])

        for log in sorted(letters):
            for id in sorted(letters[log]):
                output.append(id + ' ' + log)
        
        for log in digits:
            output.append(log)

        return output